dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteThreadPool.java

```diff
@@ -101,7 +101,8 @@ public void submitStateEvent(StateEvent stateEvent) {
      * Handle the events belong to the given workflow.
      */
     public void executeEvent(final WorkflowExecuteRunnable workflowExecuteThread) {
-        if (!workflowExecuteThread.isStart() || workflowExecuteThread.eventSize() == 0) {
+        if (!workflowExecuteThread.isStart() || workflowExecuteThread.isHandingEvent()
```

Comment:
Why we need to do this change, if the `workflowExecuteThread` is handing event, it should be filter by `if (multiThreadFilterMap.containsKey(workflowExecuteThread.getKey()))`

Reply:
When this case occur, means the system may exist bug or meet some error, like database crash.

Reply:
> When this case occur, means the system may exist bug or meet some error, like database crash.

emm. u are correct, i has avoid this change.